sklearn/neighbors/regression.py

```diff
@@ -291,9 +291,28 @@ def predict(self, X):
             y_pred = np.array([np.mean(_y[ind, :], axis=0)
                                for ind in neigh_ind])
         else:
-            y_pred = np.array([(np.average(_y[ind, :], axis=0,
-                                           weights=weights[i]))
```

Comment:
can't we just add `if len(ind) else np.nan` here?

Reply:
Right, great idea - I've changed this now.

Reply:
Hmm, on second thought I guess that it should match the second axis of _y? So inserting np.full(y_.shape[1], np.nan) under the else clause.

Reply:
Yes, that's right. Is it tested?

Reply:
Yes, I've tested it - works for one or more _y variables. Will push it now.